index.html

```diff
@@ -1261,7 +1261,7 @@ <h2>Extensibility</h2>
         </li>
       <li>
 Representations MAY define other extensibility mechanisms including methods for
-decentralized extensions. Such extension mechanisms MUST support lossless
+decentralized extensions. Such extension mechanisms SHOULD support lossless
 conversion into any other conformant representation.
```

Comment:
I would be more comfortable with this if we also say they have to tell us at least SOME way to get things into and out of the data model, even if it's just because of the default type rules. Otherwise we could end up with ambiguous behaviors due to extensions.

I'm not sure if this captures it but I'd recommend something like this:

```suggestion
decentralized extensions. Such extension mechanisms SHOULD support lossless
conversion into any other conformant representation. Extension mechanisms for a 
representation MUST define a mapping of all properties and representation syntax 
into the abstract data model and its type system.
```

Reply:
Hmm I'm wondering about cases such as CBOR where AFAIK you can have numbers as keys in a map. Would we say that a CBOR DID document representation that uses this CBOR-specific feature is not conformant, because the data model only supports property names that are strings?

@jonnycrunch interested in your opinion on this..

Reply:
CBOR where you have numbers as keys in a map is only true BEFORE consumption or AFTER production. The representation rules would state how you consume numbers-as-keys in the particular CBOR representation to the ADM, so, AFAICT, there isn't an issue here, @peacekeeper.

Reply:
The conversion rules for CBOR (not yet written) would need to explicitly say how to map member names (which can be numbers or strings) into property names (which need to be strings). Similar logic needs to be written for JSON-LD, where the value of `context` can change the semantic mapping of an object member name such that it might map to a different property name.

Reply:
@peacekeeper ... waiting on you to merge this PR. You either need to accept @jricher's suggestion, or provide an alternate that @jricher is happy with.

Reply:
I dont really agree with this. It should be allowed to have representation-specific extensions that don't fit into the ADM. In the CBOR example, I don't think it's possible to define ADM production/consumption rules that properly deal with numbers-as-keys. 

Of course such extensions that don't fit in the ADM can't be losslessly converted to other representations, but that has always been part of the model.

So I'd prefer to merge the PR without applying @jricher 's suggestion (maybe open a separate issue for that afterwards?)

Reply:
We could keep your extra sentence, but change MUST to SHOULD in that extra sentence as well.

Reply:
Waiting on @jricher to agree to @peacekeeper's suggestion, provide a different modification, etc.

Reply:
I propose to commit this suggestion, but with MUST changed to SHOULD. @jricher ?

Reply:
Applying suggestion by @peacekeeper and merging we've been waiting 18 days for a response and haven't gotten one. It's a single word change, if there are objections, a future PR can fix this easily.